File: Algorithm_169_Majority Element.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int majorityElement(vector<int>& nums)
{
	int size=nums.size(),count=0;
	if (size==1) return nums[0];
	sort(nums.begin(),nums.end());
	for (int i=0;i<size-1;i++)
	{
		if (nums[i+1]==nums[i]) count++;
		if (count>=size/2) return nums[i];
	}
}

File: Algorithm_169_Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        major, count = nums[0], 1
        for i in range(1, len(nums)):
            if count == 0:
                major, count = nums[i], 1
            else:
                count = count + 1 if nums[i] == major else count - 1
        return major